### DISCUSSION TITLE
Can't query components made in previous system

### DISCUSSION THREAD

**Discussion by eatenpancreas** [bevyengine/bevy#10139] [Q&A] [answered]
Hello, i am still relatively new to bevy, but starting to get a grip more and more. After doing a couple tutorials i decided to try out my own game: a simplified civ clone.

However, i'm currently running into a seemingly small problem that i might just not be seeing. I want to query the components made inside `owners::setup` right after inside setup::populate_grid, but the query length in populate_grid is 0. 

I have been unable to find any related discussions.

Here's the code:
```rs
App::new()
  ...
  .add_systems(Startup, (
      setup::setup,
      owners::setup.before(setup::populate_grid),
      setup::populate_grid,
  ))
  ...
```

```rs
// owners.rs
pub fn setup(
    mut commands: Commands,
    grid: Res<Grid>
) {
    let mut rng = rand::thread_rng();
    let taken_positions = vec![];
    
    for _ in 0..5 {
        let pos = (
            rng.gen_range(0..grid.0.width),
            rng.gen_range(0..grid.0.height)
        );
        
        if taken_positions.contains(&pos) { continue; }
        
        commands.spawn((
            Owner { col: random_color(&mut rng), pos }
        ));
        
        println!("Owner: {:?}", pos);
    }
}
```

```rs
// setup.rs

pub(crate) fn populate_grid(
    mut commands: Commands,
    owners: Query<&Owner>
    ...
) {
    println!("ran");
    
    println!("count= {}" , owners.iter().count());
    
    for owner in owners.iter() {
        println!("Owner results : {:?}", owner.pos);
    }
    ...
}
```

Output:
```
Owner: (23, 9)
Owner: (18, 2)
Owner: (20, 6)
Owner: (9, 8)
Owner: (12, 0)
ran
count= 0
```

Why is count 0?

**Comment by alice-i-cecile** [comment, marked as answer]
Commands aren't processed immediately. Instead, they're queued up and applied at the next `apply_deferred` synchronization point. 

For now, you'll want to insert a copy of `apply_deferred` between your systems manually, but https://github.com/bevyengine/bevy/pull/9822 will make your code as written "just work" by doing this for you automatically.

**Reply by Trashtalk217** [reply]
In other words, wait a couple weeks and update to 0.12, it will probably (fingers crossed) be fixed by then.

**Reply by alice-i-cecile** [reply]
I think we're likely to wait on 0.13 for that particular change: it's too surprising to squeeze in last minute.

**Reply by Trashtalk217** [reply]
Sorry, I misremembered the milestone it was planned for.
